tools/cli_helper.class.inc

```diff
@@ -0,0 +1,181 @@
+     *
+     * @return void
+     */
+    function printBulletPoint(string $line): void
+    {
+        echo "\t* $line\n";
+    }
+
+    /**
+     * Prints a line of text with a newline before and after. Can also be
+     * printed in a colour.
+     *
+     * @param string $line
+     * @param string $colour
+     *
+     * @return void
+     */
+    function printLine(
+        string $line,
+        string $colour = '',
+        $handle = STDOUT
+    ) {
+        $output = $line;
+        if (!empty($colour)) {
+            // Verify a valid colour is supplied
+            if (!array_key_exists(strtolower($colour), self::COLOURS)) {
+                throw InvalidArgumentException(
+                    "Invalid argument for colour (`$colour`)"
+                );
+            }
+
+            // Format the output to colour the whole line.
+            $output = sprintf("\e[%sm$line\e[0m", self::COLOURS[$colour]);
+        }
+
+        fwrite($handle, $output . PHP_EOL);
+
+        // Don't include colours in log files
+        if ($this->_loggingEnabled) {
+            file_put_contents(
+                __DIR__ . '/logs/'  . $this->_loggingFileName,
+                $line . PHP_EOL,
+                FILE_APPEND
+            );
+        }
+    }
+
+    /**
+     * Print the input formatted as a header for output.
+     *
+     * @param string $line The output.
+     *
+     * @return void
+     */
+    function printSuccess(string $line): void
+    {
+        // Takes 'input', prints '[*] input' in green text.
+        $this->printLine("[*] $line", 'green', STDOUT);
+    }
+
+    /**
+     * Print the input formatted in yellow text.
+     *
+     * @param string $line The output.
+     *
+     * @return void
+     */
+    function printWarning(string $line): void
+    {
+        // Takes 'input', prints '[-] input' in yellow text.
+        $this->printLine("[-] $line", 'yellow', STDOUT);
+    }
+    /**
+     * Print the input formatted in red text.
+     *
+     * @param string $line The output.
+     *
+     * @return void
+     */
+    public function printError(string $line): void
+    {
+        $output = "[!] $line";
+        $this->printLine($output, 'red', STDERR);
+    }
+
+    /**
+     * Returns the POSIX username for the user on the server that is
+     * executing this script.
+     */
+    public function getUsername(): string
+    {
+        return posix_getpwuid(posix_geteuid())['name'];
+    }
+
+    /**
+     * Log script output to a file.
+     *
+     * @param string $scriptName The name of the script that is using this class.
+     *
+     * @return void
+     */
+    public function enableLogging(string $scriptName): void
+    {
+        // Create a new file in the format:
+        // `$scriptName-YEAR-MONTH-DAY_HOUR-MINUTE-SECOND.log`
+        $this->_loggingFileName = sprintf(
+            "%s-%s.log",
+            // Remove extension information
+            explode('.', $scriptName)[0],
```

Comment:
This pose a problem in the case of a script named like `this.wonderfull.script.php`
Could use pathinfo()